cmd/rpcdaemon/commands/eth_api.go

```diff
@@ -324,11 +324,16 @@ func newRPCTransaction(tx types.Transaction, blockHash common.Hash, blockNumber
 	}
 	switch t := tx.(type) {
 	case *types.LegacyTx:
-		chainId = types.DeriveChainId(&t.V).ToBig()
-		result.GasPrice = (*hexutil.Big)(t.GasPrice.ToBig())
-		result.V = (*hexutil.Big)(t.V.ToBig())
-		result.R = (*hexutil.Big)(t.R.ToBig())
-		result.S = (*hexutil.Big)(t.S.ToBig())
+                if (t.V == *(uint256.NewInt(27)) || t.V == *(uint256.NewInt(28))) {
```

Comment:
1. better avoid `new` for constant things. use pre-allocated `u256.N27` 
2. I see same logic inside `types.DeriveChainId()` - why need compare with 27 again outside of `DeriveChainId`? Maybe you can point me to same logic in Geth's code?
3. We have one more code for RLP parsing of txs: https://github.com/ledgerwatch/erigon-lib/blob/main/types/txn.go#L329 - do we need some changes there also?